salt/orchestrate/aws/mitxpro_autoscaling.sls

```diff
@@ -0,0 +1,85 @@
+{% set env_settings = salt.cp.get_file_str("salt://environment_settings.yml")|load_yaml %}
+{% set ENVIRONMENT = salt.environ.get('ENVIRONMENT', 'mitxpro-production') %}
+{% set BUSINESS_UNIT = salt.environ.get('BUSINESS_UNIT', env_data.business_unit) %}
+{% set purpose = salt.grains.get('purpose', 'xpro-production') %}
+{% set env_data = env_settings.environments[ENVIRONMENT] %}
+{% set purpose_data = env_data.purposes[purpose] %}
+{% set sqs_queue = ENVIRONMENT ~ env_data.provider_services.sqs.queue %}
+{% set sns_topic = ENVIRONMENT ~ env_data.provider_services.sns.topic %}
+{% set edx_codename = purposes[PURPOSE].versions.codename %}
+
+{% set region = 'us-east-1' %}
+{% set AWS_ACCOUNT_ID = salt.vault.read('secret-operations/global/aws-account-id') %}
+{% set ami_id = salt.sdb.get('sdb://consul/edx_{}_{}_ami_id'.format(ENVIRONMENT, edx_codename)) %}
+
+create_{{ sqs_queue }}-sqs-queue:
+  boto_sqs.present:
+    - name: {{ sqs_queue }}
+    - region: {{ region }}
+    - attributes:
+        Policy: {"Version":"2012-10-17","Id":"arn:aws:sqs:{{ region }}:{{ AWS_ACCOUNT_ID }}:{{ sqs_queue }}/SQSDefaultPolicy","Statement":[{"Effect":"Allow","Principal":{"AWS":["arn:aws:iam::{{ AWS_ACCOUNT_ID }}:role/mitx-salt-master-role"]},"Action":"SQS:*","Resource":"arn:aws:sqs:{{ region }}:{{ AWS_ACCOUNT_ID }}:{{ sqs_queue }}}]}
+
+create_{{ sns_topic }}-sns-topic:
+  boto_sns.present:
+    - name: {{ sns_topic }}
+    - region: {{ region }}
+    - subscriptions:
+        - protocol: sqs
+        - endpoint: 'arn:aws:sqs:{{ region }}:{{ AWS_ACCOUNT_ID }}:{{ sqs_queue }}'
+
+create_autoscaling_group:
+  boto_asg.present:
+    - name: edx-{{ ENVIRONMENT }}-autoscaling-group
+    - launch_config:
+      - ebs_optimized: false
+      - instance_profile_name: edx-instance-role
+      - image_name: {{ ami_id }}
+      - key_name: salt-master-prod
+      - instance_type: {{ purpose_data.instances.edx.type }}
+      - security_groups:
+        - salt_master
+        - consul-agent
+        - edx
+    - min_size: {{ purpose_data.instances.edx.min_number }}
+    - max_size: {{ purpose_data.instances.edx.max_number }}
+    - desired_capacity: {{ purpose_data.instances.edx.min_number }}
+    - region: {{ region }}
+    - availability_zones:
+      - us-east-1b
+      - us-east-1c
+      - us-east-1d
+    - load_balancers:
+      - edx-{{ purpose }}-{{ ENVIRONMENT }}
+    - suspended_processes:
+        - AddToLoadBalancer
+        - AlarmNotification
+    - scaling_policies:
+        - name: ScaleUp
+          adjustment_type: ChangeInCapacity
+          as_name: edx-{{ ENVIRONMENT }}-autoscaling-group
+          cooldown: 1800
+          scaling_adjustment: 1
```

Comment:
We might want to scale up by 2 so we don't run into an issue where we're scaling up but not enough and then we have to wait for the cooldown to scale again.